File: Sorting/selection_sort.py
class Solution:
    def selectionSort(self, nums):
        for i in range(0, len(nums)-1):
            min = i
            for j in range(i+1,len(nums)):#back to work
                if(nums[j]<nums[min]):
                    min = j
            temp = nums[i]
            nums[i] = nums[min]
            nums[min] = temp
    
if __name__ == "__main__":
    nums = [64, 25, 12, 22, 11]
    print("Original array:", nums)
    Solution().selectionSort(nums)
    print("Sorted array:", nums)

File: Sorting/selection_sort.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    vector<int> selectionSort(vector<int> &nums)
    {
        for (int i = 0; i < nums.size() - 1; i++)
        {
            int min = i;
            for (int j = i + 1; j < nums.size(); j++)
            {
                if (nums[min] > nums[j])
                {
                    min = j;
                }
            }
            int temp = nums[i];
            nums[i] = nums[min];
            nums[min] = temp;
        }
        return nums;
    }
};

int main()
{
    vector<int> nums = {4, 2, 8, 6, 9};
    Solution sol;
    vector<int> sorted = sol.selectionSort(nums);

    cout << "Sorted array: ";
    for (int num : sorted)
        cout << num << " ";
    cout << endl;

    return 0;
}
